timelock-impl/src/test/java/com/palantir/atlasdb/timelock/AsyncTimelockServiceImplTest.java

```diff
@@ -19,21 +19,180 @@
+        assertThat(response.getFreshTimestamps().getStart()).isGreaterThan(response.getMinLeased());
+    }
+
+    // TODO(aalouane): enable when AsyncLockService implementation is merged
+    @Disabled
+    @ValueSource(ints = {1, 5, 10})
+    @ParameterizedTest
+    public void acquireNamedMinTimestampLeaseReturnsExactlyTheNumberOfFreshTimestampsRequested(int count) {
+        NamedMinTimestampLeaseResponse response = acquireNamedMinTimestampLease(count);
+        // This is a stronger guarantee than what we provide, but assuming the timestamp
+        // service used returns increments of 1, the tests allows us to verify we pass
+        // the right input to the timestamp service
+        assertThat(response.getFreshTimestamps().getCount()).isEqualTo(count);
+    }
+
+    // TODO(aalouane): enable when AsyncLockService implementation is merged
+    @Disabled
+    @Test
+    public void acquireNamedMinTimestampLeaseLocksWithAFreshTimestamp() {
+        timestampService.fastForwardTimestamp(10000L);
+        NamedMinTimestampLeaseResponse response = acquireNamedMinTimestampLease(10);
+        assertThat(response.getMinLeased()).isEqualTo(10001L);
+    }
+
+    // TODO(aalouane): enable when AsyncLockService implementation is merged
+    @Disabled
+    @Test
+    public void getNamedMinTimestampLeaseReturnsFreshTimestampWhenNoLeaseIsHeld() {
+        timestampService.fastForwardTimestamp(10000L);
+        assertThat(getMinLeasedNamedTimestamp()).isEqualTo(10001L);
+    }
+
+    // TODO(aalouane): enable when AsyncLockService implementation is merged
+    @Disabled
+    @Test
+    public void getNamedMinTimestampLeaseReturnsMinLeasedWhenLeaseIsHeld() {
+        // The assumption is that the timestamp service used gives incremental timestamps starting
+        // with 0 and with no gaps. Thus, a call to acquire is expected to fetch the number of
+        // timestamps requested plus one.
+
+        // For the math to make sense, we need to account for getMinLeasedNamedTimestamp grabbing
+        // a fresh timestamp before the value is fetched.
+
+        // Expected to lock with 1
+        NamedMinTimestampLeaseResponse response1 = acquireNamedMinTimestampLease(10);
+        assertThat(getMinLeasedNamedTimestamp()).isEqualTo(1);
+
+        // Expected to lock with 13
+        NamedMinTimestampLeaseResponse response2 = acquireNamedMinTimestampLease(5);
+        assertThat(getMinLeasedNamedTimestamp()).isEqualTo(1);
+
+        assertThat(lockService.unlock(response1.getLeasedLock().getLockToken())).isTrue();
+        // Expected to lock with 20
+        acquireNamedMinTimestampLease(7);
+        assertThat(getMinLeasedNamedTimestamp()).isEqualTo(13);
+
+        assertThat(lockService.unlock(response2.getLeasedLock().getLockToken())).isTrue();
+        // Expected to lock with 29
+        acquireNamedMinTimestampLease(8);
+        assertThat(getMinLeasedNamedTimestamp()).isEqualTo(20);
+    }
```

Comment:
for what it's worth, good coverage! just that I'm a bit worried about the whole "relying on contiguity" arrangement